find-modules/rules_engine.py

```diff
@@ -80,7 +84,10 @@ def __init__(self, db, rule_number, fn, pattern_zip):
         try:
             groups = ["(?P<{}>{})".format(name, pattern) for pattern, name in pattern_zip]
             groups = _SEPARATOR.join(groups)
-            self.matcher = re.compile(groups)
+            #
+            # We'll use re.match to anchor the start of the match, and so need a $ to anchor the end.
```

Comment:
The typical syntax for matching a complete line of text starts with a "^" to fix the start start of the line, and a "$" to fix the end. Now, the code in the rules_engine does not bother with the "^" because the actual matching is done with the re.match() function which implies the "^" part (circa line 151) but when I was reviewing the code, I noticed that I'd forgotten to explicitly add the "$". This change fixes that oversight.

As to what it changes, I'm not sure I expect to see many changes. The use of the _SEPARATOR.join() should delimit the fields, but there might be edge cases. For example, in matching the " const" suffix in the last part of a FunctionRuleDb would obviously also match a " constfooblah" suffix without the "$"...but that is a bit made up. A more serious case - but likely more rare - would be when using a negative match expression: I believe that "(?! const)" would unexpectedly match " const" as well without the "$"!!!